faculty/clients/job.py

```diff
@@ -194,13 +201,28 @@ class JobDefinitionSchema(BaseSchema):
     instance_size_type = fields.String(
         data_key="instanceSizeType", required=True
     )
-    instance_size = fields.Nested(
-        InstanceSizeSchema, data_key="instanceSize", required=False
-    )
+    instance_size = fields.Nested(InstanceSizeSchema, data_key="instanceSize")
     max_runtime_seconds = fields.Integer(
         data_key="maxRuntimeSeconds", required=True
     )
 
+    @validates_schema
+    def validate_instance_size(self, data):
+        if (
+            data["instance_size_type"] == "custom"
+            and data["instance_size"] is None
+        ):
+            raise ValidationError(
+                "need to specify instance size for custom instances"
+            )
+        elif (
+            data["instance_size_type"] != "custom"
+            and data["instance_size"] is not None
+        ):
+            raise ValidationError(
+                "instance_size can only be specified if instance_size_type is 'custom' "
```

Comment:
Looks like `flake8` is failing as this line is too long.

I suggest breaking the string across two lines, e.g.:

```python
            raise ValidationError(
                "instance_size can only be specified "
                "if instance_size_type is 'custom'"
            )
```